### PR TITLE
Improve aurora unlock script

### PR DESCRIPTION
**Pull request by olga24912**
Automatic extraction of all necessary data for unlock. 

Now only nonce should be provided for unlock.

### CODE REVIEW

File: aurora/scripts/utils.js [hunk 1]
```diff
@@ -168,6 +172,45 @@ async function forceIncreaseBalance(signer, config, fastBridgeAddress, token, re
   console.log("Transaction hash: ", receipt.hash);
 }
 
+async function get_pending_transfers(config, nonce) {
+    const nearAPI = require("near-api-js");
+
+    const { connect } = nearAPI;
+
+    let connectionConfig = {
```
Comment (karim-en) [inline aurora/scripts/utils.js:180]: The RPC URLs should be included in the config files

File: aurora/scripts/utils.js [hunk 2]
```diff
@@ -168,6 +172,45 @@ async function forceIncreaseBalance(signer, config, fastBridgeAddress, token, re
   console.log("Transaction hash: ", receipt.hash);
 }
 
+async function get_pending_transfers(config, nonce) {
```
Comment (karim-en) [inline aurora/scripts/utils.js:175]:
```suggestion
async function get_pending_transfer(config, nonce) {
```

This function returns just one transfer by nonce

File: aurora/hardhat.config.js [hunk 3]
```diff
@@ -101,14 +101,25 @@ task('unlock', 'Unlock tokens on Near')
...
     .setAction(async taskArgs => {
         const { unlock } = require('./scripts/utils');
         const [signer] = await hre.ethers.getSigners();
         const config = require(`./configs/${taskArgs.auroraFastBridgeConfigName}.json`);
 
-        await unlock(signer, config, taskArgs.fastBridgeAddress, taskArgs.nonce, taskArgs.ethTokenAddress, taskArgs.validTillBlockHeight);
+        await unlock(signer, config, taskArgs.fastBridgeAddress, taskArgs.nonce);
+    });
+
+task('get_pending_transfers', 'Unlock tokens on Near')
```
Comment (karim-en) [inline aurora/hardhat.config.js:112]:
```suggestion
task('get_pending_transfer', 'Get pending transfer by nonce on Near')
```
This command returns just one transfer by nonce

File: aurora/Makefile [hunk 4]
```diff
@@ -32,7 +31,7 @@ init_token_transfer:
 	yarn hardhat init_token_transfer --network $(NETWORK) --aurora-fast-bridge-config-name $(CONFIG_NAME) --fast-bridge-address $(AURORA_FAST_BRIDGE_ADDRESS) --near-token-account-id $(NEAR_TOKEN_ACCOUNT_ID) --aurora-token-address $(AURORA_TOKEN_ADDRESS) --eth-token-address $(ETH_TOKEN_ADDRESS)
 
 unlock:
-	yarn hardhat unlock --network $(NETWORK) --aurora-fast-bridge-config-name $(CONFIG_NAME) --fast-bridge-address $(AURORA_FAST_BRIDGE_ADDRESS) --nonce $(NONCE) --eth-token-address $(ETH_TOKEN_ADDRESS) --valid-till-block-height $(VALID_TILL_BLOCK_HIEGHT)
+	yarn hardhat unlock --network $(NETWORK) --aurora-fast-bridge-config-name $(CONFIG_NAME) --fast-bridge-address $(AURORA_FAST_BRIDGE_ADDRESS) --nonce $(NONCE)
```
Comment (karim-en) [inline aurora/Makefile:34]: Let's remove the `--fast-bridge-address` everywhere and use the configuration instead.

**Review by karim-en — CHANGES_REQUESTED** [review]

File: aurora/scripts/utils.js [hunk 5]
```diff
@@ -2,6 +2,7 @@ require('dotenv').config();
 const hre = require("hardhat");
 
 const {encodeInitMsgToBorsh} = require("../test/EncodeInitMsgToBorsh");
+const {getUnlockProof} = require("../test/UnlockProof");
```
Comment (karim-en) [inline aurora/scripts/utils.js:5]: There is a hardcoded provider in this function:
```
const ETH_RPC_ENDPOINT_URL = 'https://ethereum-goerli-rpc.allthatnode.com';

const provider = new ethers.getDefaultProvider(ETH_RPC_ENDPOINT_URL);
const web3 = new Web3(ETH_RPC_ENDPOINT_URL);
```

Comment (karim-en) [reply aurora/scripts/utils.js:5]: And please remove the web3 provider

**Review by karim-en — CHANGES_REQUESTED** [review]

File: aurora/hardhat.config.js (hunk 3 above)
Comment (olga24912) [reply aurora/hardhat.config.js]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/130/commits/7556ca83c35179d100ede6c99eb21bfd252f6403

File: aurora/scripts/utils.js (hunk 1 above)
Comment (olga24912) [reply aurora/scripts/utils.js]: fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/130/commits/64a50b71205c844681306d55fd96a378f1addddb

File: aurora/package.json [hunk 6]
```diff
@@ -9,11 +9,11 @@
     "@auroraisnear/aurora-sdk": "^0.0.1",
     "@nomicfoundation/hardhat-chai-matchers": "^2.0.1",
     "@nomicfoundation/hardhat-ethers": "^3.0.4",
+    "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (olga24912) [reply aurora/package.json:12]: I have no idea. 

The only thing that I have done, I install near-api-js:
```
npm i --save near-api-js
```

**Review by karim-en — APPROVED** [review]

**Pull request merged**